toolkit/scripts/check_signatures.py

```diff
@@ -20,20 +21,75 @@ def getSignature(fileName) -> str:
             sha256sum.update(read_data)
     return sha256sum.hexdigest()
 
-def check_folder(path):
+def find_file_and_check(path, filename, expected_signature) -> Optional[bool]:
+    path_to_check = os.path.join(path, filename)
+    if Path(path_to_check).is_file():
+        actual_signature = getSignature(path_to_check)
+        if actual_signature == expected_signature:
+            return True
+        print(f"ERROR: detected a mismatched signature for {filename}, expected [{expected_signature}] does not equal actual [{actual_signature}]")
+        return False
+
+    for content in os.listdir(path):
+        path_to_check = os.path.join(path, content)
+        if os.path.isdir(path_to_check):
+            result = find_file_and_check(path_to_check, filename, expected_signature)
+            if result is not None:
+                return result
+
+    return None
+
+def find_name_of_all_spec_and_signatures_json_pairs(path: str) -> List[str]:
+    names: List[str] = []
+    # Search for all spec files (XXX.spec)
+    for spec_path in Path(path).glob("*.spec"):
+        with open(spec_path, "r") as f:
+            name = Path(spec_path).stem
+            signature_path = os.path.join(path, f"{name}.signatures.json")
+            with open(signature_path, "r") as ff:
```

Comment:
There might be not signatures JSON for some spec files and `open()` will then crash.